src/main/java/net/snowflake/client/core/arrow/ArrowResultUtil.java

```diff
@@ -102,7 +102,7 @@ public static Date getDate(int day, TimeZone oldTz, TimeZone newTz) throws SFExc
    */
   private static long moveToTimeZoneOffset(
       long milliSecsSinceEpoch, TimeZone oldTZ, TimeZone newTZ) {
-    if (oldTZ.getRawOffset() == newTZ.getRawOffset()) {
+    if (oldTZ.getRawOffset() == newTZ.getRawOffset() && oldTZ.hasSameRules(newTZ)) {
```

Comment:
Now, we check offsets and also check that the rules are the same. UTC and London timezones have the same offset much of the time but different rules, which caused an offset difference of an hour for daylight savings days.

Reply:
`hasSameRules` seems compare the offset too. See https://docs.oracle.com/javase/7/docs/api/java/util/TimeZone.html#hasSameRules(java.util.TimeZone). Can you check whether `if (oldTZ.hasSameRules(newTZ))` is sufficient or not?

Reply:
Good catch! I took out the rawOffset check.